### ISSUE TITLE
build org.relaxng.datatype does not exist

### ISSUE THREAD

**Issue by geach** [validator/validator#190]
I may be building this wrong or have messed up the JAVA_HOME correctly. I used http://www.mitchleung.com/blog/2014/08/19/w3c-validator-in-ubuntu-14-04-1/ as a guide till the clone command. Then I am following the commands from https://validator.github.io/validator/#build-instructions

java -version
java version "1.7.0_91"
OpenJDK Runtime Environment (IcedTea 2.6.3) (7u91-2.6.3-0ubuntu0.14.04.1)
OpenJDK Server VM (build 24.91-b01, mixed mode)

When building I get:
sudo python ./build/build.py all
"git" submodule update --init
https://help.whatwg.org/extensions/a-rel/
https://help.whatwg.org/extensions/link-rel/
https://wiki.whatwg.org/wiki/MicrosyntaxDescriptions
https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry
https://wiki.whatwg.org/wiki/Validator.nu_alt_advice
./ant
You must set JAVA_HOME environment variable to point to
the directory where your JDK is installed.
Copying ./schema/.drivers/html5-all.rnc to ./schema/html5-all.rnc
Copying ./schema/.drivers/html5-its.rnc to ./schema/html5-its.rnc
Copying ./schema/.drivers/html5-no-microdata.rnc to ./schema/html5-no-microdata.rnc
Copying ./schema/.drivers/html5-rdfalite-w3c.rnc to ./schema/html5-rdfalite-w3c.rnc
Copying ./schema/.drivers/html5-rdfalite.rnc to ./schema/html5-rdfalite.rnc
Copying ./schema/.drivers/html5-svg-mathml.rnc to ./schema/html5-svg-mathml.rnc
Copying ./schema/.drivers/html5.rnc to ./schema/html5.rnc
Copying ./schema/.drivers/svg-xhtml5-rdf-mathml.rnc to ./schema/svg-xhtml5-rdf-mathml.rnc
Copying ./schema/.drivers/xhtml1-ruby-rdf-svg-mathml.rnc to ./schema/xhtml1-ruby-rdf-svg-mathml.rnc
Copying ./schema/.drivers/xhtml5-all.rnc to ./schema/xhtml5-all.rnc
Copying ./schema/.drivers/xhtml5-no-microdata.rnc to ./schema/xhtml5-no-microdata.rnc
Copying ./schema/.drivers/xhtml5-rdfalite-w3c.rnc to ./schema/xhtml5-rdfalite-w3c.rnc
Copying ./schema/.drivers/xhtml5-rdfalite.rnc to ./schema/xhtml5-rdfalite.rnc
Copying ./schema/.drivers/xhtml5-svg-mathml.rnc to ./schema/xhtml5-svg-mathml.rnc
Copying ./schema/.drivers/xhtml5.rnc to ./schema/xhtml5.rnc
Copying ./schema/.drivers/xhtml10 to ./schema/xhtml10
Copying ./schema/.drivers/rdf.rnc to ./schema/rdf/rdf.rnc
Removing ./schema/html5-all.rnc
Removing ./schema/html5-its.rnc
Removing ./schema/html5-no-microdata.rnc
Removing ./schema/html5-rdfalite-w3c.rnc
Removing ./schema/html5-rdfalite.rnc
Removing ./schema/html5-svg-mathml.rnc
Removing ./schema/html5.rnc
Removing ./schema/svg-xhtml5-rdf-mathml.rnc
Removing ./schema/xhtml1-ruby-rdf-svg-mathml.rnc
Removing ./schema/xhtml5-all.rnc
Removing ./schema/xhtml5-no-microdata.rnc
Removing ./schema/xhtml5-rdfalite-w3c.rnc
Removing ./schema/xhtml5-rdfalite.rnc
Removing ./schema/xhtml5-svg-mathml.rnc
Removing ./schema/xhtml5.rnc
Removing ./schema/html5/html5core.rnc
Removing ./schema/html5/html5full-no-microdata.rnc
Removing ./schema/html5/html5full-rdfa.rnc
Removing ./schema/html5/html5full-rdfalite.rnc
Removing ./schema/html5/html5full.rnc
Removing ./schema/html5/its20-html5-types.rnc
Removing ./schema/html5/its20-html5.rnc
Removing ./schema/html5/legacy.rnc
Removing ./schema/html5/xhtml5core-plus-web-forms2.rnc
Removing ./schema/html5/xhtml5core.rnc
Removing ./schema/html5/xhtml5full-html-no-microdata.rnc
Removing ./schema/html5/xhtml5full-html-rdfalite.rnc
Removing ./schema/html5/xhtml5full-html.rnc
Removing ./schema/html5/xhtml5full-xhtml-no-microdata.rnc
Removing ./schema/html5/xhtml5full-xhtml-rdfa.rnc
Removing ./schema/html5/xhtml5full-xhtml-rdfalite.rnc
Removing ./schema/html5/xhtml5full-xhtml.rnc
Removing ./schema/xhtml10
Removing ./schema/rdf
"javac" -g -nowarn -classpath "./dependencies/commons-codec-1.10.jar:./dependencies/commons-fileupload-1.3.1.jar:./dependencies/commons-io-2.4.jar:./dependencies/commons-logging-1.2.jar:./dependencies/commons-logging-1.2-adapters.jar:./dependencies/commons-logging-1.2-api.jar:./dependencies/galimatias-0.1.0.jar:./dependencies/httpcore-4.4.jar:./dependencies/httpclient-4.4.jar:./dependencies/icu4j-54.1.1.jar:./dependencies/javax.servlet-api-3.1.0.jar:./dependencies/jchardet-1.0.jar:./dependencies/jetty-http-9.2.9.v20150224.jar:./dependencies/jetty-io-9.2.9.v20150224.jar:./dependencies/jetty-security-9.2.9.v20150224.jar:./dependencies/jetty-server-9.2.9.v20150224.jar:./dependencies/jetty-servlet-9.2.9.v20150224.jar:./dependencies/jetty-servlets-9.2.9.v20150224.jar:./dependencies/jetty-util-9.2.9.v20150224.jar:./dependencies/jetty-util-ajax-9.2.9.v20150224.jar:./dependencies/log4j-1.2.17.jar:./dependencies/rhino-1.7R5.jar:./dependencies/xom-1.2.5.jar:./jing-trang/lib/saxon9.jar:./jing-trang/lib/xercesImpl.jar:./jing-trang/lib/xml-apis.jar:./jing-trang/lib/isorelax.jar" -sourcepath "./htmlparser/src" -d "./htmlparser/classes" -encoding UTF-8 -target 1.6 -source 1.6 @temp-javac-list
Removing temp-javac-list
"jar" cf "./htmlparser/dist/htmlparser.jar" @temp-jar-list
Removing temp-jar-list
Removing ./htmlparser/classes
Removing ./htmlparser/dist
"javac" -g -nowarn -d "./classes" -encoding UTF-8 -target 1.6 -source 1.6 ./src/nu/validator/xml/SaxCompiler.java
"java" -cp ./classes nu.validator.xml.SaxCompiler site/PageEmitter.xml ./src/nu/validator/servlet/PageEmitter.java
file:site/PageEmitter.xml
"java" -cp ./classes nu.validator.xml.SaxCompiler site/FormEmitter.xml ./src/nu/validator/servlet/FormEmitter.java
file:site/FormEmitter.xml
Removing ./classes
"javac" -g -nowarn -classpath "./dependencies/commons-codec-1.10.jar:./dependencies/commons-fileupload-1.3.1.jar:./dependencies/commons-io-2.4.jar:./dependencies/commons-logging-1.2.jar:./dependencies/commons-logging-1.2-adapters.jar:./dependencies/commons-logging-1.2-api.jar:./dependencies/galimatias-0.1.0.jar:./dependencies/httpcore-4.4.jar:./dependencies/httpclient-4.4.jar:./dependencies/icu4j-54.1.1.jar:./dependencies/javax.servlet-api-3.1.0.jar:./dependencies/jchardet-1.0.jar:./dependencies/jetty-http-9.2.9.v20150224.jar:./dependencies/jetty-io-9.2.9.v20150224.jar:./dependencies/jetty-security-9.2.9.v20150224.jar:./dependencies/jetty-server-9.2.9.v20150224.jar:./dependencies/jetty-servlet-9.2.9.v20150224.jar:./dependencies/jetty-servlets-9.2.9.v20150224.jar:./dependencies/jetty-util-9.2.9.v20150224.jar:./dependencies/jetty-util-ajax-9.2.9.v20150224.jar:./dependencies/log4j-1.2.17.jar:./dependencies/rhino-1.7R5.jar:./dependencies/xom-1.2.5.jar:./jing-trang/lib/saxon9.jar:./jing-trang/lib/xercesImpl.jar:./jing-trang/lib/xml-apis.jar:./jing-trang/lib/isorelax.jar:./jars/htmlparser.jar:jing-trang/build/jing.jar" -sourcepath "./src" -d "./classes" -encoding UTF-8 -target 1.6 -source 1.6 @temp-javac-list
./src/nu/validator/datatype/KeyLabelList.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/AbstractDatatype.java:29: package org.relaxng.datatype does not exist
import org.relaxng.datatype.Datatype;
                           ^
./src/nu/validator/datatype/AbstractDatatype.java:30: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/AbstractDatatype.java:31: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeStreamingValidator;
                           ^
./src/nu/validator/datatype/AbstractDatatype.java:32: package org.relaxng.datatype does not exist
import org.relaxng.datatype.ValidationContext;
                           ^
./src/nu/validator/datatype/AbstractDatatype.java:42: cannot find symbol
symbol: class Datatype
public abstract class AbstractDatatype implements Datatype {
                                                  ^
./src/nu/validator/datatype/KeyLabelList.java:42: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.KeyLabelList
      public void checkValid(CharSequence literal) throws DatatypeException {
                                                          ^
./src/nu/validator/datatype/AbstractDatatype.java:58: cannot find symbol
symbol  : class ValidationContext
location: class nu.validator.datatype.AbstractDatatype
    public final boolean isValid(String literal, ValidationContext context) {
                                                 ^
./src/nu/validator/datatype/AbstractDatatype.java:74: cannot find symbol
symbol  : class ValidationContext
location: class nu.validator.datatype.AbstractDatatype
    public void checkValid(String literal, ValidationContext context) throws DatatypeException {
                                           ^
./src/nu/validator/datatype/AbstractDatatype.java:74: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    public void checkValid(String literal, ValidationContext context) throws DatatypeException {
                                                                             ^
./src/nu/validator/datatype/AbstractDatatype.java:78: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    public abstract void checkValid(CharSequence literal) throws DatatypeException;
                                                                 ^
./src/nu/validator/datatype/AbstractDatatype.java:87: cannot find symbol
symbol  : class ValidationContext
location: class nu.validator.datatype.AbstractDatatype
            ValidationContext context) {
            ^
./src/nu/validator/datatype/AbstractDatatype.java:86: cannot find symbol
symbol  : class DatatypeStreamingValidator
location: class nu.validator.datatype.AbstractDatatype
    public DatatypeStreamingValidator createStreamingValidator(
           ^
./src/nu/validator/datatype/AbstractDatatype.java:99: cannot find symbol
symbol  : class ValidationContext
location: class nu.validator.datatype.AbstractDatatype
    public Object createValue(String literal, ValidationContext context) {
                                              ^
./src/nu/validator/datatype/AbstractDatatype.java:206: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String message) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:210: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String head, String literal, String tail) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:214: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String head, char literal, String tail) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:218: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String message) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:222: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String head, String literal, String tail) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:226: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String head, char literal, String tail) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:233: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String message, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:237: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String head, String literal, String tail, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:241: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(String head, char literal, String tail, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:245: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String message, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:249: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String head, String literal, String tail, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatatype.java:253: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatatype
    protected DatatypeException newDatatypeException(int position, String head, char literal, String tail, boolean warning) {
              ^
./src/nu/validator/datatype/AbstractDatetime.java:29: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/AbstractDatetime.java:54: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:58: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkYear(int year) throws DatatypeException {
                                            ^
./src/nu/validator/datatype/AbstractDatetime.java:67: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:78: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:84: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:108: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:113: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:126: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:131: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/AbstractDatetime.java:141: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    protected final void checkHour(String hour) throws DatatypeException {
                                                       ^
./src/nu/validator/datatype/AbstractDatetime.java:145: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkHour(int hour) throws DatatypeException {
                                            ^
./src/nu/validator/datatype/AbstractDatetime.java:151: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    protected final void checkMinute(String minute) throws DatatypeException {
                                                           ^
./src/nu/validator/datatype/AbstractDatetime.java:155: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkMinute(int minute) throws DatatypeException {
                                                ^
./src/nu/validator/datatype/AbstractDatetime.java:161: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    protected final void checkSecond(String second) throws DatatypeException {
                                                           ^
./src/nu/validator/datatype/AbstractDatetime.java:165: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkSecond(int second) throws DatatypeException {
                                                ^
./src/nu/validator/datatype/AbstractDatetime.java:171: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    protected final void checkMilliSecond(String millisecond) throws DatatypeException {
                                                                     ^
./src/nu/validator/datatype/AbstractDatetime.java:177: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkTzd(String hours, String minutes) throws DatatypeException {
                                                               ^
./src/nu/validator/datatype/AbstractDatetime.java:184: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
    private void checkTzd(int hours, int minutes) throws DatatypeException {
                                                         ^
./src/nu/validator/datatype/AbstractDatetime.java:208: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.AbstractDatetime
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/Pattern.java:29: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/Pattern.java:59: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Pattern
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/MediaQuery.java:32: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/MediaQuery.java:158: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MediaQuery
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/MediaQuery.java:175: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MediaQuery
            List<String> warnings) throws DatatypeException {
                                          ^
./src/nu/validator/datatype/MediaQuery.java:929: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MediaQuery
            String type, List<String> warnings) throws DatatypeException {
                                                       ^
./src/nu/validator/datatype/MediaQuery.java:955: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MediaQuery
    private void errNotMediaCondition(String type) throws DatatypeException {
                                                          ^
./src/nu/validator/datatype/MediaQuery.java:966: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MediaQuery
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/FloatingPointExponent.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/FloatingPointExponent.java:47: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.FloatingPointExponent
    public void checkValid(CharSequence literal) throws DatatypeException {
                                                        ^
./src/nu/validator/datatype/IriRef.java:35: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/IriRef.java:61: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.IriRef
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/IriRef.java:95: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.IriRef
    public void checkValid(CharSequence literal) throws DatatypeException {
                                                        ^
./src/nu/validator/datatype/FloatingPointExponentPositive.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/FloatingPointExponentPositive.java:47: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.FloatingPointExponentPositive
    public void checkValid(CharSequence literal) throws DatatypeException {
                                                        ^
./src/nu/validator/datatype/MetaCharset.java:26: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/MetaCharset.java:43: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.MetaCharset
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/Idrefs.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/Idrefs.java:54: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Idrefs
    public void checkValid(CharSequence literal) throws DatatypeException {
                                                        ^
./src/nu/validator/datatype/Script.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/CdoCdcPair.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/Script.java:39: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Script
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/CdoCdcPair.java:44: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.CdoCdcPair
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/DatatypeStreamingValidatorImpl.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/DatatypeStreamingValidatorImpl.java:26: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeStreamingValidator;
                           ^
./src/nu/validator/datatype/DatatypeStreamingValidatorImpl.java:29: cannot find symbol
symbol: class DatatypeStreamingValidator
        DatatypeStreamingValidator {
        ^
./src/nu/validator/datatype/DatatypeStreamingValidatorImpl.java:54: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.DatatypeStreamingValidatorImpl
    public void checkValid() throws DatatypeException {
                                    ^
./src/nu/validator/datatype/Month.java:28: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/Month.java:56: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Month
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/Month.java:61: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Month
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/Month.java:74: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.Month
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/SvgPathData.java:25: package org.relaxng.datatype does not exist
import org.relaxng.datatype.DatatypeException;
                           ^
./src/nu/validator/datatype/SvgPathData.java:57: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
            throws DatatypeException {
                   ^
./src/nu/validator/datatype/SvgPathData.java:170: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkm() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:189: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkM() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:208: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkl() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:219: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
            throws IOException, DatatypeException {
                                ^
./src/nu/validator/datatype/SvgPathData.java:255: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkL() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:266: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
            throws IOException, DatatypeException {
                                ^
./src/nu/validator/datatype/SvgPathData.java:302: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkh() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:342: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkH() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:382: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkv() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:422: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkV() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:462: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkc() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:513: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkC() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:564: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkq() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:611: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkQ() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:658: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checks() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:705: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkS() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:752: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkt() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:795: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkT() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:838: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checka() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:920: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkA() throws DatatypeException, IOException {
                                 ^
./src/nu/validator/datatype/SvgPathData.java:1002: cannot find symbol
symbol  : class DatatypeException
location: class nu.validator.datatype.SvgPathData
    private void checkArg(char command, String arg) throws DatatypeException,
                                                           ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
100 errors

Is this a JAVA_HOME issue or is there something else?

**Comment by sideshowbarker** [comment]
It’s a JAVA_HOME issue. Otherwise the following message would not be getting logged:

> You must set JAVA_HOME environment variable to point to
the directory where your JDK is installed.

If you’re literally following http://www.mitchleung.com/blog/2014/08/19/w3c-validator-in-ubuntu-14-04-1/ then I guess you need to make sure `/usr/lib/jvm/java-6-openjdk` exists. I think it likely doesn’t, since Java6 is basically obsolete at this point. It’s more likely you need to be using `/usr/lib/jvm/java-7-openjdk-amd64`.

**Comment by geach** [comment]
Found the issue. I was upgrading from unbuntu 12.02 to 14.04 both being 32 bit so the paths have i386 instead. Since I was having issues getting things to work with java 6 so installed java 7. Which gave the error of Unable to find a -javac compiler. Which only had openjdk-7-jre installed  since both packages use the same folder structure I didn't realize this at 1st. Once I installed openjdk-7-jdk things stared working as expected. So it probably was a combination of not realizing what packages were installed and upgrading the OS changing things.

Thanks for taking the time replying to something that was completely user error.
